sdk/src/wallet/utxo.ts

```diff
@@ -179,3 +181,122 @@ export function getInputFromUtxoAndTx(
 
     return input;
 }
+
+
+/**
+ * Estimate the tx inclusion fee for a given address or public key with an optional OP_RETURN output.
+ *
+ * @param fromAddress The Bitcoin address which is sending to the `toAddress`.
+ * @param amount The amount of BTC (as satoshis) to send. If no amount is specified, the fee is estimated for all UTXOs, i.e., the max amount.
+ * @param publicKey Optional public key needed if using P2SH-P2WPKH.
+ * @param opReturnData Optional OP_RETURN data to include in an output.
+ * @param confirmationTarget The number of blocks to include this tx (for fee estimation).
+ * @param utxoSelectionStrategy The strategy to use for selecting UTXOs. See https://github.com/paulmillr/scure-btc-signer/tree/main#utxo-selection for options.
+ * @returns {Promise<number>} The fee amount for estiamted transaction inclusion in satoshis.
+ * 
+ * @dev Wtih no amount set, we estimate the fee for all UTXOs by trying to spend all inputs using strategy 'all'. If an amount is set, we use the 'default
+ * strategy to select the UTXOs.
+ */
+export async function estimateTxFee(
+    fromAddress: string,
+    amount?: number,
+    publicKey?: string,
+    opReturnData?: string,
+    confirmationTarget: number = 3,
+    utxoSelectionStrategy: string = 'default'
+): Promise<number> {
```

Comment:
If the feerate changes between the tx estimation and the submission of an actual tx, then we can still get failures. Maybe we should either take feerate as an argument (as the [old function](https://github.com/bob-collective/bob/blob/13e08e27b9f2d770f0655c86ecb149809f4cd425/sdk/src/utils.ts#L198) did, or include it in the return value?